src/plugins/initiatives/initiative.js

```diff
@@ -0,0 +1,28 @@
+// @flow
+
+import {type NodeAddressT} from "../../core/graph";
+
+/**
+ * This makes the assumption a Champion cannot fail in championing.
+ * Instead of a success status, they should be removed if unsuccessful.
+ *
+ * There is also no timestamp for completion or each edge.
+ * They will all use the initiative creation date instead.
+ * We can support accurate edge timestamps by interpreting wiki histories.
+ * However the additional complexity and requirements put on the tracker
+ * don't seem worthwhile right now.
+ * Especially because cred can flow even before bounties are released.
+ * See https://discourse.sourcecred.io/t/write-the-initiatives-plugin/269/6
+ * This will likely create dangling edges, as the initiative may be created
+ * before some of it's contributions/references/dependencies exist as a node.
+ */
+export type Initiative = {|
+  +title: string,
+  +timestampMs: number,
+  +completed: boolean,
+  +tracker: NodeAddressT,
+  +dependencies: $ReadOnlyArray<NodeAddressT>,
```

Comment:
Great point. I was going to depend on the reference detection part for this. Testing independent of that wouldn't be too difficult as I was planning to make an HTML => links function and test that. Suppose we cached this format though, like for the mirror plugins. Having the addresses would mean the schema depends on the implementation of reference detection of other plugins :sweat_smile:

Going with URLs for the intermediate, and doing a just-in-time reference detection when creating the graph seems better.

Reply:
Changed this.

Reply:
I did go back to `tracker: NodeAddressT` though. As I believe this will be known before we're even trying to parse data from the tracker. The alternative I was thinking of was to set it to the `TopicId` type. However this doesn't include the server URL.